File: src/main/java/toast/client/modules/combat/AutoRespawn.java
package toast.client.modules.combat;

import net.minecraft.client.gui.screen.DeathScreen;
import toast.client.event.EventImpl;
import toast.client.event.events.player.EventUpdate;
import toast.client.modules.Module;
import toast.client.utils.TimerUtil;

public class AutoRespawn extends Module {

    private TimerUtil timer = new TimerUtil();

    public AutoRespawn() {
        super("AutoRespawn", Category.COMBAT, -1);
        this.addNumberOption("Speed", 2, 1, 20);
    }

    public void onEnable() {
       timer.reset();
    }

    public void onDisable() {
        timer.reset();
    }

    @EventImpl
    public void onEvent(EventUpdate event) {
        if(mc.currentScreen != null) {
            if(this.timer.isDelayComplete((long) (this.getDouble("Speed") * 1000L))) {
                this.timer.setLastMS();
                if(mc.currentScreen instanceof DeathScreen) {
                    mc.player.requestRespawn();
                    mc.openScreen(null);
                }
            }
        }
    }
}

File: src/main/java/toast/client/modules/combat/AutoRespawn.kt
package toast.client.modules.combat

import com.google.common.eventbus.Subscribe
import net.minecraft.client.gui.screen.DeathScreen
import toast.client.events.network.EventSyncedUpdate
import toast.client.modules.Module
import toast.client.utils.TimerUtil

/**
 * Module to automatically press the respawn button when the player dies
 */
class AutoRespawn : Module(
        "AutoRespawn",
        "Automatically presses the respawn button for you.",
        Category.COMBAT,
        -1
) {
    private val timer = TimerUtil()
    override fun onEnable() {
        timer.reset()
    }

    override fun onDisable() {
        timer.reset()
    }

    @Subscribe
    fun onEvent(event: EventSyncedUpdate?) {
        if (timer.isDelayComplete((getDouble("Speed").times(1000L)))) {
            timer.reset()
            if (mc.currentScreen ?: return is DeathScreen) {
                (mc.player ?: return).requestRespawn()
                mc.openScreen(null)
            }
        }
    }

    init {
        settings.addSlider("Speed", 1.0, 2.0, 20.0)
    }
}